autorest/codegen/models/code_model.py

```diff
@@ -284,6 +280,14 @@ def add_inheritance_to_models(self) -> None:
         self._add_properties_from_inheritance()
         self._add_exceptions_from_inheritance()
 
+    def _populate_target_property(self, parameter: Parameter) -> None:
+        for obj in self.sorted_schemas:
+            for prop in obj.properties:
+                if prop.id == parameter.target_property_name:
+                    parameter.target_property_name = prop.name
```

Comment:
Would have prefered a link to the property, in case we need type or anyting like that later, but we can refactor if necessary (not blocking)